packages/next/build/utils.ts

```diff
@@ -593,10 +607,11 @@ export async function isPageStatic(
 
     const config = mod.config || {}
     return {
-      static: !hasStaticProps && !hasGetInitialProps,
+      static: !hasStaticProps && !hasGetInitialProps && !hasServerProps,
```

Comment:
Can we rename this in a different PR to be `isStatic` or `hasNoDataRequirement`?

Reply:
Opened PR here: https://github.com/zeit/next.js/pull/10260